platform/posix/transport/src/sockets_posix.c

```diff
@@ -143,29 +160,91 @@ static SocketStatus_t resolveHostName( const char * pHostName,
 }
 /*-----------------------------------------------------------*/
 
+static SocketStatus_t connectToAddress( struct sockaddr * pAddrInfo,
+                                        const char * pHostName,
+                                        size_t hostNameLength,
+                                        uint16_t port,
+                                        int tcpSocket )
+{
+    SocketStatus_t returnStatus = SOCKETS_SUCCESS;
+    int connectStatus = 0;
+    char resolvedIpAddr[ INET6_ADDRSTRLEN ];
+    socklen_t addrInfoLength;
+    uint16_t netPort = 0;
+
+    assert( pAddrInfo != NULL );
+    assert( pHostName != NULL );
+    assert( hostNameLength > 0 );
+
+    /* Convert port from host byte order to network byte order. */
+    netPort = htons( port );
+
+    if( pAddrInfo->sa_family == AF_INET )
```

Comment:
Since the return values of inet_ntop() are ignored, seems like we need an assert on sa_family.